### PR TITLE
User requests, car requests, events requests, errorHandlers, auth

### PR DESCRIPTION
**Pull request by cstruchevs**

### CODE REVIEW

File: app.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+User.hasMany(Car, {foreignKey: "user_id", onDelete: "CASCADE", onUpdate:"CASCADE"})
+User.belongsToMany(Event, {through: EventParticipants})
+Event.belongsToMany(User, {through: EventParticipants})
+
+const port = process.env.PORT || 5000;
+
+const start = (): void => {
+  sequelize
+    .sync({ force: true })
+    //.sync()
```
Comment (VolodymyrDevico) [inline app.ts:45]: remove

File: controllers/authConroller.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  //await user.hashPassword();
+
+  const token = user.createJWT();
+  res.status(StatusCodes.CREATED).json({
+    user,
+    token,
+  });
+};
+
+export const login: RequestHandler = async (req, res) => {
```
Comment (VolodymyrDevico) [inline controllers/authConroller.ts:33]: split on different folders

File: controllers/authConroller.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  const { email, password } = req.body;
+  if (!email || !password) {
+    throw new BadRequestError("Please provide all values");
```
Comment (VolodymyrDevico) [inline controllers/authConroller.ts:36]: Some better error text

File: controllers/dist/eventController.js [hunk 4]
```diff
@@ -0,0 +1,187 @@
+"use strict";
```
Comment (VolodymyrDevico) [inline controllers/dist/eventController.js:1]: gitignore

File: db/database.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { Sequelize } from "sequelize";
+
+const sequelize = new Sequelize('devico_project', 'root', 'semen2012', {
```
Comment (VolodymyrDevico) [inline db/database.ts:3]: move to .env

File: errors/not-found.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import { StatusCodes } from 'http-status-codes'
+import CustomAPIError from './custom-api'
+
+class NotFoundError extends CustomAPIError {
+    public statusCude = StatusCodes.NOT_FOUND
```
Comment (VolodymyrDevico) [inline errors/not-found.ts:5]: add prettier

File: middleware/error-handler.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    statusCode: err.statusCode || StatusCodes.INTERNAL_SERVER_ERROR,
+    msg: err.message || 'Something went wrong, try again later',
+  }
+  if (err.name === 'ValidationError') {
+    defaultError.statusCode = StatusCodes.BAD_REQUEST
+    defaultError.msg = Object.values(err.errors)
+      .map((item:any) => item.message)
+      .join(',')
+  }
+  if (err.code && err.code === 11000) {
```
Comment (VolodymyrDevico) [inline middleware/error-handler.ts:20]: Create constants with good naming

**Review by VolodymyrDevico — CHANGES_REQUESTED** [review]

**Review by turchin-aa — COMMENTED** [review]
Gj

**Pull request closed without merging**